ts/Extensions/Drilldown.ts

```diff
@@ -1633,3 +1634,59 @@ addEvent(Point, 'afterSetState', function (): void {
         applyCursorCSS(this.series.halo, 'auto', false, styledMode);
     }
 });
+
+addEvent(H.Chart, 'beforeShowResetZoom', function (): void {
+    if (this.drillUpButton) {
+        this.temporaryDrillUpButton = this.drillUpButton;
+        delete this.drillUpButton;
+    }
+}, {
+    order: 0
+});
+
+addEvent(H.Chart, 'afterShowResetZoom', function (): void {
+    if (this.temporaryDrillUpButton && this.resetZoomButton) {
+        var buttonOptions = this.options.drilldown && this.options.drilldown.drillUpButton,
+            bbox = this.resetZoomButton.getBBox();
+
+        if (buttonOptions && buttonOptions.position && buttonOptions.position.x) {
+            this.temporaryDrillUpButton.align({
+                x: buttonOptions.position.x - bbox.width - 20,
+                y: buttonOptions.position.y,
+                align: 'right'
+            },
+            false,
+            buttonOptions.relativeTo || 'plotBox'
+            );
+        }
+
+        this.drillUpButton = this.temporaryDrillUpButton;
+        delete this.temporaryDrillUpButton;
+    }
+});
+
+addEvent(H.Chart, 'selection', function (event: any): void {
+    if (event.resetSelection === true && this.drillUpButton) {
+        var buttonOptions = this.options.drilldown && this.options.drilldown.drillUpButton;
+
+        if (buttonOptions && buttonOptions.position) {
+            this.drillUpButton.align({
+                x: buttonOptions.position.x,
+                y: buttonOptions.position.y,
+                align: 'right'
+            },
+            false,
+            buttonOptions.relativeTo || 'plotBox'
+            );
+        }
+    }
```

Comment:
Lines 1668-1681 are exactly the same as 1649-1661. I suggest to extract it and avoid code rendudant.